### PR TITLE
tweak Vec.map/mapi

### PR DESCRIPTION
**Pull request by Lampese**
The behavior of map/mapi is not normal in Vec, they are not immutable.

At the same time, I would like to inquire whether the map implemented by this inplace should be implemented?

### CODE REVIEW

File: vec/vec_test.mbt [hunk 1]
```diff
@@ -149,14 +149,20 @@ test "clear" {
...
 test "map" {
   let v = Vec::[3, 4, 5]
-  v.map(fn(x) { x + 1 })
-  @assertion.assert_eq(v, Vec::[4, 5, 6])?
+  let e : Vec[Int] = Vec::[]
+  inspect(v.map(fn(x) { x + 1 }), content="Vec::[4, 5, 6]")?
+  inspect(v.map(fn(x) { x.to_string() }), content="Vec::[3, 4, 5]")?
+  inspect(e.map(fn(x) { x + 1 }), content="Vec::[]")?
+  // Is it correct?
+  inspect(Vec::["a", "b", "c"], content="Vec::[a, b, c]")?
```
Comment (Lampese) [inline vec/vec_test.mbt:157]:
```
// Is it correct?
inspect(Vec::["a", "b", "c"], content="Vec::[a, b, c]")?
```

Comment (bobzhang) [reply vec/vec_test.mbt:157]: this is a known issue in current derived Show. cc @Guest0x0

**Review by bobzhang — APPROVED** [review]

**Comment by bobzhang** [comment]
>  I would like to inquire whether the map implemented by this inplace should be implemented?

Do you have a name for that?

**Comment by Lampese** [comment]
> >  I would like to inquire whether the map implemented by this inplace should be implemented?
> 
> Do you have a name for that?

I have two options.
- map_inplace
- map_mutable

**Comment by bobzhang** [comment]
map_inplace lgtm

**Pull request merged**